### PR TITLE
[Keyboard] Add Olly JF Rev.2

### PR DESCRIPTION
**Pull request by mechlovin**
## Description
Add support for Olly JF Rev. 2 PCB (APM32F103, IS31FL3731 mono LED).
Thanks.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by lendunistus** [comment]
from the PR checklist:

`* keyboard moves within the repository must go through the develop branch instead of master, so as to ensure compatibility for users
     * data/mappings/keyboard_aliases.hjson must be updated to reflect the move, so users with pre-created configurator keymap.json files continue to detect the correct keyboard`

File: keyboards/mechlovin/olly/jf/rev2/config.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#define ENABLE_LED_MATRIX_WAVE_UP_DOWN	
+#define ENABLE_LED_MATRIX_SOLID_REACTIVE_SIMPLE	
+#define ENABLE_LED_MATRIX_SOLID_REACTIVE_WIDE	
+#define ENABLE_LED_MATRIX_SOLID_REACTIVE_MULTIWIDE	
+#define ENABLE_LED_MATRIX_SOLID_REACTIVE_CROSS	
+#define ENABLE_LED_MATRIX_SOLID_REACTIVE_MULTICROSS	
+#define ENABLE_LED_MATRIX_SOLID_REACTIVE_NEXUS	
+#define ENABLE_LED_MATRIX_SOLID_REACTIVE_MULTINEXUS	
+#define ENABLE_LED_MATRIX_SOLID_SPLASH	
+#define ENABLE_LED_MATRIX_SOLID_MULTISPLASH
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/config.h:55]: can these settings be migrated to `led_matrix.animations` in `info.json`
```suggestion
```

File: keyboards/mechlovin/olly/jf/rev1/config.h
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/config.h:1]: can this setting be moved to `matrix_pins.rows` in `info.json`, in turn removing his file

File: keyboards/mechlovin/olly/jf/rev1/info.json [hunk 2]
```diff
@@ -0,0 +1,663 @@
+{
+    "keyboard_name": "Olly JF Rev1",
+    "usb": {
+        "pid": "0xD180",
+        "device_version": "0.0.1"
+    },
+    "backlight": {
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/info.json:7]: migrated from `rules.mk`
```suggestion
    "backlight": {
        "enabled": true,
```

File: keyboards/mechlovin/olly/jf/rev1/info.json [hunk 3]
```diff
@@ -0,0 +1,663 @@
+{
+    "keyboard_name": "Olly JF Rev1",
+    "usb": {
+        "pid": "0xD180",
+        "device_version": "0.0.1"
+    },
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/info.json:6]: migrated from `rules.mk`
```suggestion
    },
    "matrix_pins": {
        "custom_lite": true
    },
```

File: keyboards/mechlovin/olly/jf/rev1/rules.mk [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Processor frequency
+F_CPU = 16000000
+
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+CUSTOM_MATRIX = lite
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/rules.mk:5]: migrated to `info.json`
```suggestion
```

File: keyboards/mechlovin/olly/jf/rev1/readme.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Mechlovin Olly JF
+
+![jf](https://i.imgur.com/IrfTGE7l.png)
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/readme.md:3]:
```suggestion
![jf](https://i.imgur.com/IrfTGE7h.png)
```

File: keyboards/mechlovin/olly/jf/rev1/readme.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+    make mechlovin/olly/jf:default
+
+Flashing example for this keyboard:
+
+    make mechlovin/olly/jf:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+**Reset Key:** 3 ways to put the Olly JF into bootloader:
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/readme.md:21]:
```suggestion
## Bootloader

Enter the bootloader in 3 ways:

```

File: keyboards/mechlovin/olly/jf/rev2/readme.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Mechlovin Olly jf/rev2
+
+![jf/rev2](https://i.imgur.com/IrfTGE7l.png)
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/readme.md:3]:
```suggestion
![jf/rev2](https://i.imgur.com/IrfTGE7h.png)
```

File: keyboards/mechlovin/olly/jf/rev2/readme.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+    make mechlovin/olly/jf/rev2:default
+
+Flashing example for this keyboard:
+
+    make mechlovin/olly/jf/rev2:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+**Reset Key:** 3 ways to put the Olly jf/rev2 into bootloader:
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/readme.md:21]:
```suggestion
## Bootloader

Enter the bootloader in 3 ways:

```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/default/keymap.c:51]: 1. removed non-functional layers
2. no means stated to access layers 1-3
```suggestion
    )
```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/default_ansi/keymap.c [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                         _______, _______, _______, _______, 
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______,             _______, _______, _______,    _______, _______, _______
+    ),
+
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/default_ansi/keymap.c:55]: 1. removed non-functional layers
2. no means stated to access layers 1-3
```suggestion
    )
```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/default_ansi_split_bs_rshift/keymap.c [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                         _______, _______, _______, _______, 
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______,             _______, _______, _______,    _______, _______, _______
+    ),
+
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/default_ansi_split_bs_rshift/keymap.c:55]: 1. removed non-functional layers
2. no means stated to access layers 1-3
```suggestion
    )
```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/default_iso/keymap.c [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                         _______, _______, _______, _______, 
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______,             _______, _______, _______,    _______, _______, _______
+    ),
+
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/default_iso/keymap.c:55]: 1. removed non-functional layers
2. no means stated to access layers 1-3
```suggestion
    )
```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/default_iso_split_bs_rshift/keymap.c [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______,    _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                         _______, _______, _______, _______, 
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,    _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______,             _______, _______, _______,    _______, _______, _______
+    ),
+
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/default_iso_split_bs_rshift/keymap.c:55]: 1. removed non-functional layers
2. no means stated to access layers 1-3
```suggestion
    )
```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,  KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/via/keymap.c:35]: 1. removed non-functional layers
2. no means stated to access layer 1
```suggestion
    )
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/config.h
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/config.h:1]: please move contents of this file into `/rev1/config.h`

File: keyboards/mechlovin/olly/jf/rev1/config.h [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define MATRIX_ROW_PINS { D5, D6, A5, A4, A3, A6}
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/config.h:20]: please move MATRIX_ROW_PINS into `matrix_pins.rows` in `rev1/info.json`

File: keyboards/mechlovin/olly/jf/info.json [hunk 16]
```diff
@@ -1,686 +1,40 @@
...
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "rgblight": true
     },
-    "indicators": {
-        "caps_lock": "B0",
-        "num_lock": "B1",
-        "scroll_lock": "B2"
+    "matrix_pins": {
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/info.json:17]: please move `matrix_pins.rows` into `rev1/info.json`

File: keyboards/mechlovin/olly/jf/rev2/info.json [hunk 17]
```diff
@@ -0,0 +1,605 @@
...
+        "nkro": true,
+        "command": true,
+        "console": true,
+        "led_matrix": true
+    },
+    "ws2812": {
+        "pin": "B5",
+        "driver": "spi"
+    },
+    "led_matrix": {
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/info.json:21]: `led_matrix.animations` is missing

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/rev1/config.h
Comment (drashna) [inline keyboards/mechlovin/olly/jf/rev1/config.h:1]: This file can be removed.

File: keyboards/mechlovin/olly/jf/rev2/config.h [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+#pragma once
+
+// This is a 7-bit address, that gets left-shifted and bit 0
+// set to 0 for write, 1 for read (as per I2C protocol)
+// The address will vary depending on your wiring:
+// 0b1110100 AD <-> GND
+// 0b1110111 AD <-> VCC
+// 0b1110101 AD <-> SCL
+// 0b1110110 AD <-> SDA
+#define LED_DRIVER_ADDR_1 0b1110110
```
Comment (drashna) [inline keyboards/mechlovin/olly/jf/rev2/config.h:27]: There are some changes on develop to how these are structured/set.  
```suggestion
// 00 AD <-> GND
// 01 AD <-> SCL
// 10 AD <-> SDA
// 11 AD <-> VCC
// ADDR represents A1:A0 of the 7-bit address.
// The result is: 0b11101(ADDR)
#define LED_DRIVER_ADDR_1 IS31FL3731_I2C_ADDRESS_SDA
```

File: keyboards/mechlovin/olly/jf/rev2/config.h [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+#define I2C_DRIVER I2CD2
+#define I2C1_SCL_PIN B10
+#define I2C1_SDA_PIN B11
+#define LED_DRIVER_COUNT 1
+#define LED_MATRIX_LED_COUNT 114
+#define LED_MATRIX_KEYPRESSES // reacts to keypresses
+#define LED_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
```
Comment (drashna) [inline keyboards/mechlovin/olly/jf/rev2/config.h:34]: Only one of these will take precedence.

File: keyboards/mechlovin/olly/jf/rev2/rev2.c [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+    { 0, C1_5  },  { 0, C2_5 },  { 0, C3_5 },  { 0, C4_5  },  { 0, C5_5 },  { 0, C6_5  },  { 0, C7_5 },  { 0, C8_5  },  { 0, C9_5 },  { 0, C1_12 },  { 0, C2_12 },  { 0, C3_12  },  { 0, C4_12 },  { 0, C5_12  },                 { 0, C7_12  },  { 0, C8_12 },  { 0, C9_12 },  { 0, C4_16 },
+    { 0, C1_4  },  { 0, C2_4 },  { 0, C3_4 },  { 0, C4_4  },  { 0, C5_4 },  { 0, C6_4  },  { 0, C7_4 },  { 0, C8_4  },  { 0, C9_4 },  { 0, C1_13 },  { 0, C2_13 },  { 0, C3_13  },  { 0, C4_13 },  { 0, C5_13  },  { 0, C6_13 },  { 0, C7_13  },  { 0, C8_13 },  { 0, C9_13 },  { 0, C5_16 },
+    { 0, C1_3  },  { 0, C2_3 },                { 0, C4_3  },                               { 0, C7_3 },                                              { 0, C2_14 },  { 0, C3_14  },  { 0, C4_14 },  { 0, C5_14  },  { 0, C6_14 },  { 0, C7_14  },  { 0, C8_15 },  { 0, C9_15 },  { 0, C7_16 },
+    { 0, C2_1  },  { 0, C1_1 },  { 0, C9_1 },  // Lock LED Indicator
+    { 0, C7_15 },  { 0, C6_15 }, { 0, C9_16 }, { 0, C8_16 }, { 0, C1_15 }, // Layer LED Indicator
+};
+
+led_config_t g_led_config = {
```
Comment (drashna) [inline keyboards/mechlovin/olly/jf/rev2/rev2.c:37]: I believe that this can be migrated to the info.json, as well.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/rev1/info.json [hunk 21]
```diff
@@ -0,0 +1,668 @@
...
+    "matrix_pins": {
+        "custom_lite": true,
+        "rows": ["D5", "D6", "A5", "A4", "A3", "A6"]
+    },
+    "backlight": {
+        "enabled": true,
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/info.json:12]:
```suggestion
    },
    "features": {
        "backlight": true
    },
    "matrix_pins": {
        "custom_lite": true,
        "rows": ["D5", "D6", "A5", "A4", "A3", "A6"]
    },
    "backlight": {
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/rev2/rev2.c [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+    if (host_keyboard_led_state().scroll_lock) {
+        led_matrix_set_value(108, 0xFF);
+    } else {
+        led_matrix_set_value(108, 0x00);
+    }
+    return true;
+}
+
+
+layer_state_t layer_state_set_kb(layer_state_t state) {
```
Comment (drashna) [inline keyboards/mechlovin/olly/jf/rev2/rev2.c:88]: The code here should also be in led_matrix_indicators_kb, as well. 

If there is an animation enabled, these values would get wiped out.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/rev2/rev2.c (hunk 22 above)
Comment (mechlovin) [reply keyboards/mechlovin/olly/jf/rev2/rev2.c]: okay.

File: keyboards/mechlovin/olly/jf/config.h [hunk 23]
```diff
@@ -21,18 +21,3 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 /* key matrix size */
 #define MATRIX_ROWS 6
 #define MATRIX_COLS 19
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/config.h:23]: please move these two settings to a, new, `config.h` file inside `keyboards/mechlovin/olly/jf/rev1` directory
```suggestion
```

File: keyboards/mechlovin/olly/jf/rev1/readme.md [hunk 24]
```diff
@@ -0,0 +1,27 @@
+# Mechlovin Olly JF
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/readme.md:1]:
```suggestion
# Mechlovin Olly JF Rev. 1
```

File: keyboards/mechlovin/olly/jf/rev1/readme.md [hunk 25]
```diff
@@ -0,0 +1,27 @@
...
+* Hardware Supported: Jetfire Korean Kustom
+* Hardware Availability: [Mechlovin' Studio](https://mechlovin.studio/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make mechlovin/olly/jf:default
+
+Flashing example for this keyboard:
+
+    make mechlovin/olly/jf:default:flash
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev1/readme.md:17]:
```suggestion
    make mechlovin/olly/jf/rev1:default

Flashing example for this keyboard:

    make mechlovin/olly/jf/rev1:default:flash
```

File: keyboards/mechlovin/olly/jf/rules.mk [hunk 26]
```diff
@@ -1,18 +1 @@
...
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
-NKRO_ENABLE = no            # Enable N-Key Rollover
-BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
-AUDIO_ENABLE = no           # Audio output
-CUSTOM_MATRIX = lite
-
-SRC += matrix.c
+DEFAULT_FOLDER = mechlovin/olly/jf/rev1
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rules.mk:1]: add EoF
```suggestion
DEFAULT_FOLDER = mechlovin/olly/jf/rev1

```

File: keyboards/mechlovin/olly/jf/rev2/rules.mk [hunk 27]
```diff
@@ -0,0 +1 @@
+# This file intentionally left blank
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/rules.mk:1]: add EoF
```suggestion
# This file intentionally left blank

```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/via/keymap.c [hunk 28]
```diff
@@ -0,0 +1,27 @@
...
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                    KC_F13,           KC_HOME, KC_END,  KC_PGUP, KC_PGDN,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  KC_INS,           KC_NUM,  KC_PSLS, KC_PAST, KC_PAUS,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_DEL,           KC_P7,   KC_P8,   KC_P9,   KC_PMNS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,                             KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+        KC_LSFT, KC_SPC,  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_RSFT,          KC_UP,            KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT,          KC_P0,   KC_PDOT, KC_PENT
+    )
+};
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/via/keymap.c:27]: add EoF
```suggestion
};

```

File: keyboards/mechlovin/olly/jf/rev2/keymaps/default/keymap.c [hunk 29]
```diff
@@ -0,0 +1,27 @@
...
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                    KC_F13,           KC_HOME, KC_END,  KC_PGUP, KC_PGDN,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  KC_INS,           KC_NUM,  KC_PSLS, KC_PAST, KC_PAUS,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_DEL,           KC_P7,   KC_P8,   KC_P9,   KC_PMNS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,           KC_ENT,                            KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_RSFT,          KC_UP,            KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT,          KC_P0,   KC_PDOT, KC_PENT
+    )
+};
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/default/keymap.c:27]: add EoF
```suggestion
};

```

File: keyboards/mechlovin/olly/jf/rev2/config.h [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+#define LED_MATRIX_FRAMEBUFFER_EFFECTS // enable framebuffer effects
+
+#define I2C1_CLOCK_SPEED 400000
+#define I2C1_DUTY_CYCLE FAST_DUTY_CYCLE_2
+
+#define WS2812_SPI SPID1 // default: SPID1
+#define WS2812_SPI_MOSI_PAL_MODE 0 // MOSI pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 5
+
+#define WEAR_LEVELING_BACKING_SIZE 4096
+#define WEAR_LEVELING_LOGICAL_SIZE 2048
```
Comment (dunk2k) [inline keyboards/mechlovin/olly/jf/rev2/config.h:46]: add EoF
```suggestion
#define WEAR_LEVELING_LOGICAL_SIZE 2048

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by dunk2k** [comment]
Does `keyboards/olly/jf/matrix_diagram.md` apply to both rev1 and rev2?

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
This has some config errors that are preventing it from compiling.  These need addressing.

**Comment by mechlovin** [comment]
> Does `keyboards/olly/jf/matrix_diagram.md` apply to both rev1 and rev2? It appears, from layout information, that aforementioned file is applicable to rev1 only. Could it be moved into rev1 directory and have rev1 appended to title?

Hi, It just for rev.1. I will move to rev.1. 
Thank you.

**Comment by mechlovin** [comment]
> This has some config errors that are preventing it from compiling. These need addressing.

Solved, Thank you.

File: keyboards/mechlovin/olly/jf/rev2/config.h [hunk 31]
```diff
@@ -0,0 +1,43 @@
...
+// 10 AD <-> SDA
+// 11 AD <-> VCC
+// ADDR represents A1:A0 of the 7-bit address.
+// The result is: 0b11101(ADDR)
+#define LED_DRIVER_ADDR_1 IS31FL3731_I2C_ADDRESS_SDA
+#define I2C_DRIVER I2CD2
+#define I2C1_SCL_PIN B10
+#define I2C1_SDA_PIN B11
+#define LED_DRIVER_COUNT 1
+#define LED_MATRIX_LED_COUNT 114
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/rev2/config.h:34]:
```suggestion
#define IS31FL3731_I2C_ADDRESS_1 IS31FL3731_I2C_ADDRESS_SDA

#define I2C_DRIVER I2CD2
#define I2C1_SCL_PIN B10
#define I2C1_SDA_PIN B11

```

File: keyboards/mechlovin/olly/jf/rev2/rev2.c [hunk 32]
```diff
@@ -0,0 +1,92 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+void board_init(void) {
+    AFIO->MAPR |= AFIO_MAPR_SPI1_REMAP;
+}
+
+const is31_led PROGMEM g_is31_leds[LED_MATRIX_LED_COUNT] = {
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/rev2/rev2.c:22]:
```suggestion
const is31fl3731_led_t PROGMEM g_is31fl3731_leds[IS31FL3731_LED_COUNT] = {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/rev2/keymaps/default/readme.md
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/rev2/keymaps/default/readme.md:1]: This file can be deleted.

File: keyboards/mechlovin/olly/jf/rev2/config.h [hunk 33]
```diff
@@ -17,22 +17,18 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
- *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
- *
- */
-#define MATRIX_ROW_PINS { D5, D6, A5, A4, A3, A6}
-
-/* COL2ROW, ROW2COL */
-#define DIODE_DIRECTION ROW2COL
+#define LED_MATRIX_KEYPRESSES // reacts to keypresses
+#define LED_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
+#define LED_MATRIX_FRAMEBUFFER_EFFECTS // enable framebuffer effects
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/rev2/config.h:28]:
```suggestion
```

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**